_test_yaml/test/src/build_config.g.dart

```diff
@@ -36,35 +34,29 @@ Map<String, dynamic> _$ConfigToJson(Config instance) => <String, dynamic>{
     };
 
 T _$enumDecode<T>(
-  Map<T, dynamic> enumValues,
-  dynamic source, {
-  T unknownValue,
+  Map<T, Object> enumValues,
+  Object? source, {
+  T? unknownValue,
 }) {
   if (source == null) {
-    throw ArgumentError('A value must be provided. Supported values: '
-        '${enumValues.values.join(', ')}');
+    throw ArgumentError(
+      'A value must be provided. Supported values: '
+      '${enumValues.values.join(', ')}',
+    );
   }
 
   final value = enumValues.entries
-      .singleWhere((e) => e.value == source, orElse: () => null)
+      .cast<MapEntry<T, Object>?>()
```

Comment:
Might want to work on a better solution for this... for instance you could eagerly throw the argument error inside of `orElse` and not need the cast?

Reply:
Hrm...good idea!